src/state/middleware/scigateway.middleware.tsx

```diff
@@ -147,6 +150,17 @@ const ScigatewayMiddleware: Middleware = ((
     return store.dispatch(requestPluginRerender());
   }
 
+  if (action.type === LoadDarkModePreferenceType) {
+    next(action);
+
+    const darkModeLocalStorage = localStorage.getItem('darkMode');
+    const darkModePreference = darkModeLocalStorage === 'true' ? true : false;
+    const theme = buildTheme(darkModePreference);
```

Comment:
Surely if the `action` is of type `LoadDarkModePreferenceType` you can just get the preference from `action.payload.preference` rather than querying localstorage?